agw/main.tf

```diff
@@ -0,0 +1,78 @@
+locals {
+  resource_group = "${var.environment}_${var.application_name}_gateway"
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = "${local.resource_group}"
+  location = "${var.location}"
+}
+
+resource "azurerm_subnet" "subnet" {
```

Comment:
Wouldn't this also be passed in from the subnet sub-module, is there any reason the apgw submodule needs to manage these resources?

Reply:
I was going to put in subnet first, and in fact did.  I moved it back because it only apply to the app gateway, and it seemed silly to pass it around from module to module for virtually no reason.  I can move it if you insist, but I believe this is a better more simple location for it.